### PR TITLE
Feature/avi/19 user controller, DB context

### PR DESCRIPTION
**Pull request by Avishkar-Motheelal**
Scaffold the DB context
Create user controller and services with CRUD operations
Cleanup dummy Weather Forecast files

### CODE REVIEW

File: .gitignore
Comment (JesseNaidoo) [inline .gitignore:1]: Could you maybe add try catch block in each function in the controller, to catch any exception that may occur in EFCore/Service.

Just catch the exception and return a status code of 500, with the exception Message.

File: dev-do-list-server/Controllers/UserController.cs
Comment (JesseNaidoo) [inline dev-do-list-server/Controllers/UserController.cs:1]: Could you maybe add try catch block in each function in the controller, to catch any exception that may occur in EFCore/Service.

Just catch the exception and return a status code of 500, with the exception Message.

**Comment by Michael-Rolle-BBD** [comment]
I feel like to make things quicker and easier to implement for all our other services we should create a generic service and extend it for the CRUD operations which we can inherit from for our other services. These videos explain it quite well:
- https://app.pluralsight.com/ilx/video-courses/e656b81b-e83e-4eff-8366-62609c06ae20/9a1fde8a-0c19-46b3-a32b-7819792dbe0d/090b63b2-087f-42b4-a73f-651ee23f8eaa
- https://app.pluralsight.com/ilx/video-courses/e656b81b-e83e-4eff-8366-62609c06ae20/9a1fde8a-0c19-46b3-a32b-7819792dbe0d/fab0681f-6be5-44f9-b42a-e6349c3908a5

**Comment by JesseNaidoo** [comment]
> I feel like to make things quicker and easier to implement for all our other services we should create a generic service and extend it for the CRUD operations which we can inherit from for our other services. These videos explain it quite well:
> 
> * https://app.pluralsight.com/ilx/video-courses/e656b81b-e83e-4eff-8366-62609c06ae20/9a1fde8a-0c19-46b3-a32b-7819792dbe0d/090b63b2-087f-42b4-a73f-651ee23f8eaa
> * https://app.pluralsight.com/ilx/video-courses/e656b81b-e83e-4eff-8366-62609c06ae20/9a1fde8a-0c19-46b3-a32b-7819792dbe0d/fab0681f-6be5-44f9-b42a-e6349c3908a5

I am happy with this. It will help a lot in the long run.

**Review by JesseNaidoo — APPROVED** [review]

**Pull request merged**